restli-tools/src/main/java/com/linkedin/restli/tools/snapshot/check/PegasusSchemaSnapshotCompatibilityChecker.java

```diff
@@ -0,0 +1,297 @@
+  private static String _resolverPath;
+
+  static
+  {
+    _options.addOption(OptionBuilder.withLongOpt("help")
+        .withDescription("Print help")
+        .create('h'));
+    _options.addOption(OptionBuilder.withArgName("compatibility_level")
+        .withLongOpt("compatLevel")
+        .hasArg()
+        .withDescription("Compatibility level " + listCompatLevelOptions())
+        .create("cl"));
+    _options.addOption(OptionBuilder.withArgName("compatibilityOption_mode")
+        .withLongOpt("compatMode")
+        .hasArg()
+        .withDescription("CompatibilityOption Mode " + listCompatModeOptions())
+        .create("cm"));
+    _options.addOption(OptionBuilder.withLongOpt("report")
+        .withDescription("Prints a report at the end of the execution that can be parsed for reporting to other tools")
+        .create("report"));
+  }
+
+  public static void main(String[] args) throws Exception
+  {
+    final CommandLineParser parser = new GnuParser();
+    CommandLine cl = parser.parse(_options, args);
+
+    if (cl.hasOption('h'))
+    {
+      help();
+      System.exit(0);
+    }
+
+    String[] cliArgs = cl.getArgs();
+
+    if (cliArgs.length < 2 || cliArgs.length %2 != 0)
+    {
+      _logger.error("Invalid arguments");
+      help();
+      System.exit(1);
+    }
+
+    final String compatLevelValue;
+    if (cl.hasOption("cl"))
+    {
+      compatLevelValue = cl.getOptionValue("cl");
+    }
+    else
+    {
+      compatLevelValue = CompatibilityLevel.DEFAULT.name();
+    }
+
+    CompatibilityLevel compatLevel = null;
+    try
+    {
+      compatLevel = CompatibilityLevel.valueOf(compatLevelValue.toUpperCase());
+    }
+    catch (IllegalArgumentException e)
+    {
+      _logger.error("Invalid compatibilityLevel");
+      help();
+      System.exit(1);
+    }
+
+    String compatModeValue;
+    if (cl.hasOption("cm"))
+    {
+      compatModeValue = cl.getOptionValue("cm");
+    }
+    else
+    {
+      compatModeValue = CompatibilityOptions.Mode.SCHEMA.name();
+    }
+    CompatibilityOptions.Mode compatMode = null;
+    try
+    {
+      compatMode = CompatibilityOptions.Mode.valueOf(compatModeValue.toUpperCase());
+    }
+    catch (IllegalArgumentException e)
+    {
+      _logger.error("Invalid compatibilityOption Mode");
+      help();
+      System.exit(1);
+    }
+
+
+    try
+    {
+      _resolverPath = RestLiToolsUtils.getResolverPathFromSystemProperty();
+    }
+    catch (IOException e)
+    {
+      _logger.error("Cannot read resolver path: " + e.getMessage());
+      System.exit(1);
+    }
+
+    PegasusSchemaSnapshotCompatibilityChecker compatibilityChecker = new PegasusSchemaSnapshotCompatibilityChecker();
+    for (int i = 1; i < cliArgs.length; i += 2)
+    {
+      String prevSnapshot = cliArgs[i-1];
+      String currentSnapshot = cliArgs[i];
+      compatibilityChecker.checkPegasusSchemaCompatibility(prevSnapshot, currentSnapshot, compatLevel, compatMode);
```

Comment:
Would it be cleaner if the action `checkPegasusSchemaCompatibility` return the `InfoMap` directly?
I mean : not use the check to have another call to get the info.

The reason being, if we not return info map directly, they are sort of internal state in the checker, that means you will need to get the info map before you can use the check to check again.

Reply:
Updated based on the suggestion